pytorch_lightning/trainer/trainer.py

```diff
@@ -908,6 +911,9 @@ def test(self, model=None):
         """
         self.testing = True
         if model is not None:
-            self.fit(model)
+            eval_results = self.run_evaluation(test=True, model=model)
         else:
-            self.run_evaluation(test=True)
+            eval_results = self.run_evaluation(test=True)
+        self.testing = False
```

Comment:
not sure if this is needed...

Reply:
Without setting `self.testing=False` any subsequent call to `Trainer.fit()` will act as `Trainer.test()`.

Reply:
per comment above, if you return it always, you do not need to swap it there and back...

Reply:
@Borda Thx a lot for your time. I return it always now. However, I still not understand why I shouldn't change `self.testing` back and forth.

Reply:
Sure, pls check #770

Reply:
Changes proposed by #770 are really good. I guess when you merge #770 to master, current PR will be irrelevant!

Reply:
cool! let's wait it is in master and stay tune in meantime ;]